app/models/issue_number_lookup.rb

```
@@ -0,0 +1,75 @@
+    return unless issue_nums.any? # Don't answer if we don't see #123
+    # Error messages if the project info is not set up correctly
+    return "No ProjectChannel configured for this channel." unless channel.present?
+    return "No ProjectRepo configured for this channel." if repos.empty?
+    # response_lines should never be empty at this point-- we have a repo,
+    # and we have some issue numbers, and whether the issue # is found in the
+    # repo or not, we will have some sort of message (link, or "not found")
+    return if response_lines.empty?
+    # Send back the issue info
+    response_lines.join("\n")
+  end
+
+  # Find all instances of #123 pattern in the text and return them as an array
+  # like ['#123', '#456', #'789']
+  def issue_nums
+    @_issue_nums ||= params['text'].scan(ISSUE_NUMBER_PATTERN)
+  end
+
+  # The ProjectChannel for the channel on which the message was received.
+  # Used to determine Project and therefore Repos
+  def channel
+    @_channel ||= ProjectChannel.find_by(channel_name: params['channel_name'])
+  end
+
+  # List of repos connected to this channel through the project config
+  def repos
+    @_repos ||= channel.project.repos
+  end
+
+  # Return an array of all issue search results (either title and link or "not
+  # found")
+  def response_lines
+    issue_nums.flat_map do |issue_num|
+      found = false
+      messages = repos.map do |repo|
+        msg = message_for_issue(repo, issue_num)
+        found = found || msg.present?
+        msg
+      end
+
+      if found
+        messages
+      else
+        issue_not_found_message(issue_num)
+      end
+    end.compact
+  end
+
+  # Make a link for an issue if it is found, nil otherwise
+  def message_for_issue(repo, issue_num)
+    begin
+      issue = Issue.new(org: repo.org, repo: repo.repo, number: issue_num[1..-1])
+      "<#{issue.html_url}|Issue #{repo.org}/#{repo.repo}/#{issue_num}: #{issue.title}>"
+    rescue Github::Error::NotFound
```

Do not suppress exceptions.